mosquitto/Dockerfile

```diff
@@ -0,0 +1,126 @@
+
+# Get the fork of OQS-OpenSSL_1_1_1-stable
+WORKDIR $SOURCE_PATH
+RUN git clone --depth 1 --branch OQS-OpenSSL_1_1_1-stable https://github.com/open-quantum-safe/openssl.git OQS-OpenSSL
+
+# Get and build liboqs, then install it into a subdirectory inside the OQS-OpenSSL folder
+RUN git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs.git liboqs && \
+    cd liboqs && mkdir build && cd build && \
+    cmake -GNinja $LIBOQS_BUILD_DEFINES -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=$SOURCE_PATH/OQS-OpenSSL/oqs .. && \
+    ninja && ninja install && echo "liboqs installed successfully" || exit 1
+
+# Build and install OQS-OpenSSL_1_1_1-stable
+WORKDIR $SOURCE_PATH/OQS-OpenSSL
+## OS type: x86_64, aarch64
+RUN ./Configure shared linux-x86_64 -lm --prefix=$OPENSSL_LIB_PATH/ \
+        --openssldir=$OPENSSL_LIB_PATH/ $OPENSSL_BUILD_DEFINES && \
+        make -j$(nproc) && make install && echo "OQS-OpenSSL installed successfully" || exit 1
+# RUN ./Configure shared no-asm linux-aarch64 -lm --prefix=$OPENSSL_LIB_PATH/ \
+#         --openssldir=$OPENSSL_LIB_PATH/ $OPENSSL_BUILD_DEFINES && \
+#         make -j$(nproc) && make install && echo "OQS-OpenSSL installed successfully" || exit 1
+
+# Build and install Mosquitto
+WORKDIR $SOURCE_PATH
+RUN git clone -b master https://github.com/eclipse/mosquitto.git mosquitto && cd mosquitto && \
+    make -j$(nproc) && make install && echo "Mosquitto installed successfully" || exit 1
+
+
+# Second stage: Only create minimal image:
+
+FROM ubuntu:20.04
+# FROM debian:buster
+
+ARG OPENSSL_LIB_PATH
+ARG SIG_ALG
+ENV SIG_ALG=${SIG_ALG}
+ARG BROKER_IP
+ENV BROKER_IP=${BROKER_IP}
+ARG PUB_IP
+ENV PUB_IP=${PUB_IP}
+ARG SUB_IP
+ENV SUB_IP=${SUB_IP}
+ARG EXAMPLE
+ENV EXAMPLE=${EXAMPLE}
+
+# Set the TLS_DEFAULT_GROUPS environment variable to permit selection of QSC KEMs(https://github.com/open-quantum-safe/openssl#build-options)
+ENV TLS_DEFAULT_GROUPS="kyber512"
+
+# Copy files from the local storage to a destination in the Docker image
+WORKDIR /
+RUN mkdir test
+ADD . /test
+RUN chmod 777 /test/* && sed -i 's/\r//' /test/*
+
+# Only keep the necessary library contents in the final image
+# OQS-OpenSSL
+COPY --from=builder $OPENSSL_LIB_PATH  $OPENSSL_LIB_PATH
+# Mosquitto
+COPY --from=builder /usr/local/lib  /usr/local/lib
+COPY --from=builder /usr/local/bin  /usr/local/bin
+COPY --from=builder /usr/local/sbin  /usr/local/sbin
+COPY --from=builder /lib/x86_64-linux-gnu/libcjson.so.1 /lib/x86_64-linux-gnu
+
+# Dynamically link to mosquitto
+RUN ln -s /usr/local/lib/libmosquitto.so.1 /usr/lib/libmosquitto.so.1 && ldconfig
+
+# Dynamically link to OQS-OpenSSL library
+ENV LD_LIBRARY_PATH=$OPENSSL_LIB_PATH/lib
+
+# Set path 
+ENV PATH="/usr/local/bin:/usr/local/sbin:$OPENSSL_LIB_PATH/bin:$PATH"
+
+# Generate the CA key and the cert(default algorithm: dilithium5)
+RUN openssl req -x509 -new -newkey dilithium5 -keyout /test/CA.key -out /test/CA.crt -nodes -subj "/O=test-ca" -days 3650
```

Comment:
> Shouldn't this use of dilithium5 not be rather a use of the docker build variable SIG_ALG? The latter otherwise is not used.

I hope that the security level of the CA certificate could be higher, so I use 'dilithium5' by default at first. I changed to use the docker build variable SIG_ALG.

The build variable SIG_ALG is used in three shell scripts to generate keys and certificates for server and client.